### PR TITLE
Property Based Testing

### PR DESCRIPTION
**Pull request by maksymhubenko**

### CODE REVIEW

File: src/test/java/url/shortener/server/service/impl/UrlServiceImplTestPropertyBased.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  @Test
+  void createUrlPBT() {
+    final String dummyId = "0";
+
+    qt().forAll(
+        strings().basicLatinAlphabet().ofLengthBetween(0, 10)
+    ).check(alias -> {
+      try {
+        UrlCreateDto urlCreateDto = new UrlCreateDto()
+            .setUri(new URI("https://www.google.com/"))
```
Comment (dvasyab) [inline src/test/java/url/shortener/server/service/impl/UrlServiceImplTestPropertyBased.java:54]: Working from Windows can cause some problems, such as "this methods are not resolved in UrlCreateDto". This should be fixed in future, but as for now be aware of this.

**Comment by dvasyab** [comment]
Also, Windows, some times can't delete temp files, but it works perfectly fine with Linux.

File: src/test/java/url/shortener/server/service/impl/UrlServiceImplTestPropertyBased.java (hunk 1 above)
Comment (maksymhubenko) [reply src/test/java/url/shortener/server/service/impl/UrlServiceImplTestPropertyBased.java:54]: Confirm.

**Review by dvasyab — APPROVED** [review]
Everything seems good to me.

**Review by yugisu — APPROVED** [review]

**Pull request merged**